src/main/java/org/apache/commons/collections4/bloomfilter/BitCountProducer.java

```diff
@@ -47,7 +72,7 @@ default boolean forEachIndex(IntPredicate predicate) {
 
     /**
      * Creates a BitCountProducer from an IndexProducer.  The resulting
-     * producer will count each enabled bit once.
+     * producer will return every index from the IndexProducer with a count of 1.
```

Comment:
I was considering why do we need to do this: convert an IndexProducer to a BitCountProducer. It would be to add it to a CountingBloomFilter. In this case the conversion does not remove duplicate indices. This means that the BitCountProducer can increment the count of an index more than 1 when adding a single IndexProducer.

This has the following consequences:
- The count at each index of a CountingBloomFilter does not represent the number of individual hashes added to the filter that had that index enabled. The count is bounded by the number of hashes as a lower limit, i.e. it may be higher.
- Removal of an IndexProducer from a counting bloom filter must use the same method to construct the BitCountProducer as the addition in order to use the exact same index->count mapping.

I am fine with this behaviour. Filtering an IndexProducer to unique indices is extra work. The `IndexFilter` class helps with this but requires a `Shape`. So it cannot be done in the BitCountProducer interface which has no Shape; other mechanisms to filter the IndexProducer without a bound on the index would be inefficient.

So perhaps we require a bit of extra documentation here to serve as a user-beware warning:
```
 * <p>Note that the BitCountProducer does not remove duplicates. Any use of the
 * BitCountProducer to create an aggregate mapping of index to counts, such as a
 * CountingBloomFilter, should use the same BitCountProducer in both add and
 * subtract operations to maintain consistency.
```
I believe this is consistent with the new wording in the javadoc header for the BitCountProducer interface.